### ISSUE TITLE
feature request: support for -O and -o options in bcftools reheader

### ISSUE THREAD

**Issue by wkretzsch** [samtools/bcftools#93]
This would be practical and consistent to have.

**Comment by freeseek** [comment]
According to https://samtools.github.io/bcftools/bcftools.html#reheader the "-s, --samples FILE" takes a file with "old_name new_name\n" pairs separated by whitespaces. However, what would happen if the names in the original VCF file have spaces? We have some VCF files formatted that way (unfortunately). Since the VCF format separates sample names with "\t" characters, wouldn't it be more consistent for these pairs to be formatted as "old_name\tnew_name\n"?

**Comment by pd3** [comment]
@freeseek Added a support for this (https://github.com/pd3/bcftools/commit/169d700769e0acffd1df71faa26a6a3217fe1487) in a slightly different way: both spaces and tabs in the samples file continue to function as delimiters, but spaces can be escaped using a backslash:

      Not\ a\ good\ name   NewName

This is for backward compatibility but also to give more flexibility for the more common use case. 

Please next time open a new issue so that we do not mix topics and close resolved things. Thanks.

**Comment by mcshane** [comment]
From the original issue, the `-o` option is already added, while I think we are not going to implement the `-O` option, instead requiring streaming the output through view instead (let us know if there is a case against this)
```
bcftools reheader -h new.header orig.vcf.gz | bcftools view -Ob -o new.bcf
```
The follow up issue was closed via 611297e88188c776021af612776364ee5e62b14b.
